client/plots/DEanalysis.js

```diff
@@ -208,34 +292,68 @@ add:
 	if (mavb[0].adjusted_p_value != undefined) {
 		// enable pvalue switching between adjusted and unadjusted
 		const row = holder.append('div').style('margin', '20px')
-		row.append('span').text('Select P value for Volcano plot:')
-		const select = row
-			.append('select')
-			.style('margin-left', '5px')
-			.on('change', event => {
-				minlogpv = 0
-				maxlogpv = 0
-				const useun = select.node().selectedIndex == 0
-				for (const d of mavb) {
-					const pv = useun ? d.adjusted_p_value : d.original_p_value
-					if (pv == 0) continue
-					minlogpv = Math.min(minlogpv, pv)
-					maxlogpv = Math.max(maxlogpv, pv)
-				}
-				yscale.domain([minlogpv, maxlogpv])
-				client.axisstyle({
-					axis: yaxisg.call(d3axis.axisLeft().scale(yscale)),
-					color: 'black',
-					showline: true
-				})
-				dotg.attr('transform', d => {
-					const pv = useun ? d.adjusted_p_value : d.original_p_value
-					return 'translate(' + xscale(d.fold_change) + ',' + yscale(pv) + ')'
-				})
-				ylab.text(useun ? '-log10(adjusted P value)' : '-log10(original P value)')
+		minlogpv = 0
+		maxlogpv = 0
+		let text_string
+		for (const d of mavb) {
+			let pv
+			if (p_value_adjusted_original == 'adjusted') {
+				pv = d.adjusted_p_value
+			} else {
+				pv = d.original_p_value
+			}
+			if (pv == 0) continue
+			minlogpv = Math.min(minlogpv, pv)
+			maxlogpv = Math.max(maxlogpv, pv)
+		}
+		yscale.domain([minlogpv, maxlogpv])
+		client.axisstyle({
+			axis: yaxisg.call(d3axis.axisLeft().scale(yscale)),
+			color: 'black',
+			showline: true
+		})
+		dotg.attr('transform', d => {
+			let pv
+			if (p_value_adjusted_original == 'adjusted') {
+				pv = d.adjusted_p_value
+			} else {
+				pv = d.original_p_value
+			}
+			return 'translate(' + xscale(d.fold_change) + ',' + yscale(pv) + ')'
+		})
+		if (p_value_adjusted_original == 'adjusted') {
+			text_string = '-log10(adjusted P value)'
+		} else {
+			text_string = '-log10(original P value)'
+		}
+		ylab.text(text_string)
+		holder
+			.append('div')
+			.html(
+				'Percentage of significant genes:' +
```

Comment:
This legend can be showed besides the plot instead of below, so it is visible on the screen without having to scroll.